### ISSUE TITLE
revive:  panic: Invalid argument to the context-as-argument rule. Unrecognized key allowtypesbefore

### ISSUE THREAD

**Issue by dongjiang1989** [golangci/golangci-lint#5310]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

```bash
$ golangci-lint run
panic: Invalid argument to the context-as-argument rule. Unrecognized key allowtypesbefore

goroutine 15862 [running]:
github.com/mgechev/revive/rule.getAllowTypesFromArguments({0x14001b3c940?, 0x140025e6b20?, 0x12d421d08?})
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:97 +0x328
github.com/mgechev/revive/rule.(*ContextAsArgumentRule).configure(...)
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:37
github.com/mgechev/revive/rule.(*ContextAsArgumentRule).Apply.func1()
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:21 +0x30
sync.(*Once).doSlow(0x14001a05bf8?, 0x104cc2e10?)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/sync/once.go:76 +0xf8
sync.(*Once).Do(...)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/sync/once.go:67
github.com/mgechev/revive/rule.(*ContextAsArgumentRule).Apply(0x106004210, 0x14006afc600, {0x14001b3c940?, 0x13?, 0x2d1?})
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:21 +0x6c
github.com/mgechev/revive/lint.(*File).lint(0x14006afc600, {0x14001e4e000, 0x15, 0x20}, {0x0, 0x3fe999999999999a, {0x104e45753, 0x7}, 0x0, 0x14002f10990, ...}, ...)
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/lint/file.go:108 +0x17c
github.com/mgechev/revive/lint.(*Package).lint.func1(0x140025e6a10?)
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/lint/package.go:191 +0x70
created by github.com/mgechev/revive/lint.(*Package).lint in goroutine 15858
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/lint/package.go:190 +0xb4
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.63.4 built with go1.23.3 from (unknown, modified: ?, mod sum: "h1:bJQFQ3hSfUto597dkL7ipDzOxsGEpiWdLiZ359OWOBI=") on (unknown)
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```
ref: https://github.com/prometheus/prometheus/blob/main/.golangci.yml

</details>

### Go environment

<details>

```console
$  go version && go env
go version go1.23.3 darwin/arm64
GO111MODULE='on'
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/dongjiang/Library/Caches/go-build'
GOENV='/Users/dongjiang/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/dongjiang/Documents/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/dongjiang/Documents/go'
GOPRIVATE=''
GOPROXY='https://goproxy.cn,direct'
GOROOT='/opt/homebrew/Cellar/go/1.23.3/libexec'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='local'
GOTOOLDIR='/opt/homebrew/Cellar/go/1.23.3/libexec/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.23.3'
GODEBUG=''
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/dongjiang/Library/Application Support/go/telemetry'
GCCGO='gccgo'
GOARM64='v8.0'
AR='ar'
CC='cc'
CXX='c++'
CGO_ENABLED='1'
GOMOD='/Users/dongjiang/Documents/go/src/github.com/prometheus/prometheus/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/wb/t6q4mr6d5bj96h5clzfc6xzc0000gn/T/go-build1493643372=/tmp/go-build -gno-record-gcc-switches -fno-common'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO golangci-lint has version v1.63.4 built with go1.23.3 from (unknown, modified: ?, mod sum: "h1:bJQFQ3hSfUto597dkL7ipDzOxsGEpiWdLiZ359OWOBI=") on (unknown) 
INFO [config_reader] Config search paths: [./ /Users/dongjiang/Documents/go/src/github.com/prometheus/prometheus /Users/dongjiang/Documents/go/src/github.com/prometheus /Users/dongjiang/Documents/go/src/github.com /Users/dongjiang/Documents/go/src /Users/dongjiang/Documents/go /Users/dongjiang/Documents /Users/dongjiang /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 25 linters: [depguard errcheck errorlint exptostd gocritic godot gofumpt goimports gosimple govet ineffassign loggercheck misspell nilnesserr nolintlint perfsprint predeclared revive sloglint staticcheck testifylint unconvert unused usestdlibvars whitespace] 
INFO [loader] Go packages loading at mode 8767 (types_sizes|exports_file|files|name|compiled_files|deps|imports) took 1.35268525s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 90.496167ms 
panic: Invalid argument to the context-as-argument rule. Unrecognized key allowtypesbefore

goroutine 14471 [running]:
github.com/mgechev/revive/rule.getAllowTypesFromArguments({0x140025d8f20?, 0x14000600808?, 0x14003a0ecc0?})
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:97 +0x328
github.com/mgechev/revive/rule.(*ContextAsArgumentRule).configure(...)
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:37
github.com/mgechev/revive/rule.(*ContextAsArgumentRule).Apply.func1()
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:21 +0x30
sync.(*Once).doSlow(0x1400c85ebf8?, 0x101382e10?)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/sync/once.go:76 +0xf8
sync.(*Once).Do(...)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/sync/once.go:67
github.com/mgechev/revive/rule.(*ContextAsArgumentRule).Apply(0x1026c4210, 0x14003a0ec80, {0x140025d8f20?, 0x13?, 0x2d1?})
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/rule/context_as_argument.go:21 +0x6c
github.com/mgechev/revive/lint.(*File).lint(0x14003a0ec80, {0x14002050800, 0x15, 0x20}, {0x0, 0x3fe999999999999a, {0x101505753, 0x7}, 0x0, 0x140017af620, ...}, ...)
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/lint/file.go:108 +0x17c
github.com/mgechev/revive/lint.(*Package).lint.func1(0x1400a4e7740?)
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/lint/package.go:191 +0x70
created by github.com/mgechev/revive/lint.(*Package).lint in goroutine 14712
	/Users/dongjiang/Documents/go/pkg/mod/github.com/mgechev/revive@v1.5.1/lint/package.go:190 +0xb4
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)
